btor/src/boolector_factory.cpp

```diff
@@ -7,7 +7,7 @@ namespace smt {
 /* BoolectorSolverFactory implementation */
 SmtSolver BoolectorSolverFactory::create()
 {
-  return std::make_unique<BoolectorSolver>();
+  return std::make_shared<BoolectorSolver>();
```

Comment:
👎 There doesn't seem to be an obvious reason for changing this to `shared_ptr`; `unique_ptr` is in C++11, just `make_unique` isn't and you could fix that by simply adding something like

```
template<typename T, typename ...Args>
std::unique_ptr<T> smt_switch_make_unique(Args&&... args) {
  return std::unique_ptr<T>{new T(std::forward<Args>(args)...)};
}
```

Reply:
Thanks for the feedback: it's being switched to `make_shared` because `SmtSolver` is a `shared_ptr` not `unique_ptr` in master now.